src/amo/withExperiment.js

```diff
@@ -175,15 +188,73 @@ export const withExperiment =
         WrappedComponent,
       )})`;
 
-      componentDidMount() {
-        const { _getVariant, _isExperimentEnabled, cookies, id, variants } =
-          this.props;
-
+      initialExperimentSetup(props) {
+        const {
+          _getVariant,
+          _isExperimentEnabled,
+          dispatch,
+          id,
+          storedVariant,
+          variants,
+        } = props;
+
+        let variant = null;
         const isEnabled = _isExperimentEnabled({ _config, id });
         const registeredExperiments = this.getExperiments();
         const experimentInCookie = this.cookieIncludesExperiment(
           registeredExperiments,
         );
+        const addExperimentToCookie = !experimentInCookie;
+
+        if (isEnabled) {
+          // Use the variant in the cookie if one exists, otherwise use the
+          // variant from the Redux store.
+          if (experimentInCookie) {
+            variant = registeredExperiments[id];
+          } else if (storedVariant) {
+            variant = storedVariant.variant;
```

Comment:
This only supports one experiment at a time currently, which I think is probably fine for this iteration. Maybe I should still check `storedVariant.id` here, to make sure it matches the current experiment's `id`?